presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeAggregations.java

```diff
@@ -295,6 +296,16 @@ public void testMarkDistinct()
         assertQuery("SELECT orderkey, count(distinct comment), sum(distinct linenumber) FROM lineitem GROUP BY 1");
     }
 
+    @Test
+    public void testDistinct()
+    {
+        Session session = Session.builder(getSession())
+                .setSystemProperty("use_mark_distinct", "falze")
```

Comment:
> typo.

The logic that parses session property values uses Boolean.valueOf(value), which happens to work with "falze" because it

```
Returns a Boolean with a value represented by the specified string. The Boolean returned represents a true value if the string argument is not null and is equal, ignoring case, to the string "true".
```

> Did this system property make any difference?

Yes. The query plans are very different.

With use_mark_distinct:

```
 Fragment 0 [SINGLE]
     Output layout: [count, count_2]
     Output partitioning: SINGLE []
     Stage Execution Strategy: UNGROUPED_EXECUTION
     - Output[_col0, _col1] => [count:bigint, count_2:bigint]
             _col0 := count (1:35)
             _col1 := count_2 (1:61)
         - Aggregate(FINAL) => [count:bigint, count_2:bigint]
                 count := "presto.default.count"((count_12)) (1:35)
                 count_2 := "presto.default.count"((count_13)) (1:61)
             - LocalExchange[SINGLE] () => [count_13:bigint, count_12:bigint]
                 - RemoteSource[1] => [count_13:bigint, count_12:bigint]

 Fragment 1 [HASH]
     Output layout: [count_13, count_12]
     Output partitioning: SINGLE []
     Stage Execution Strategy: UNGROUPED_EXECUTION
     - Aggregate(PARTIAL) => [count_13:bigint, count_12:bigint]
             count_13 := "presto.default.count"((linenumber)) (mask = linenumber_11) (1:61)
             count_12 := "presto.default.count"((orderkey)) (mask = orderkey_10) (1:35)
         - MarkDistinct[distinct=linenumber:integer marker=linenumber_11][$hashvalue] => [orderkey:bigint, linenumber:integer, orderkey_10:boolean, $hashvalue:bigint, linenumber_11:boolean]
             - LocalExchange[HASH][$hashvalue] (linenumber) => [orderkey:bigint, linenumber:integer, orderkey_10:boolean, $hashvalue:bigint]
                 - RemoteSource[2] => [orderkey:bigint, linenumber:integer, orderkey_10:boolean, $hashvalue_14:bigint]

 Fragment 2 [tpch:orders:15000]
     Output layout: [orderkey, linenumber, orderkey_10, $hashvalue_15]
     Output partitioning: HASH [linenumber][$hashvalue_15]
     Stage Execution Strategy: UNGROUPED_EXECUTION
     - Project[projectLocality = LOCAL] => [orderkey:bigint, linenumber:integer, orderkey_10:boolean, $hashvalue_15:bigint]
             $hashvalue_15 := combine_hash(BIGINT'0', COALESCE($operator$hash_code(linenumber), BIGINT'0')) (1:93)
         - MarkDistinct[distinct=orderkey:bigint marker=orderkey_10] => [orderkey:bigint, linenumber:integer, orderkey_10:boolean]
             - TableScan[TableHandle {connectorId='tpch', connectorHandle='lineitem:sf0.01', layout='Optional[lineitem:sf0.01]'}, grouped = false] => [orderkey:bigint, linenumber:integer]
                     Estimates: {rows: 60175 (1.43MB), cpu: 842450.00, memory: 0.00, network: 0.00}
                     orderkey := tpch:orderkey (1:93)
                     linenumber := tpch:linenumber (1:93)
```

Without:

```
 Fragment 0 [SINGLE]
     Output layout: [count, count_2]
     Output partitioning: SINGLE []
     Stage Execution Strategy: UNGROUPED_EXECUTION
     - Output[_col0, _col1] => [count:bigint, count_2:bigint]
             Estimates: {rows: 1 (18B), cpu: 1684900.00, memory: 18.00, network: 842450.00}
             _col0 := count (1:35)
             _col1 := count_2 (1:61)
         - Aggregate => [count:bigint, count_2:bigint]
                 Estimates: {rows: 1 (18B), cpu: 1684900.00, memory: 18.00, network: 842450.00}
                 count := "presto.default.count"(DISTINCT (orderkey)) (1:35)
                 count_2 := "presto.default.count"(DISTINCT (linenumber)) (1:61)
             - LocalExchange[SINGLE] () => [orderkey:bigint, linenumber:integer]
                     Estimates: {rows: 60175 (1.03MB), cpu: 842450.00, memory: 0.00, network: 842450.00}
                 - RemoteSource[1] => [orderkey:bigint, linenumber:integer]

 Fragment 1 [tpch:orders:15000]
     Output layout: [orderkey, linenumber]
     Output partitioning: SINGLE []
     Stage Execution Strategy: UNGROUPED_EXECUTION
     - TableScan[TableHandle {connectorId='tpch', connectorHandle='lineitem:sf0.01', layout='Optional[lineitem:sf0.01]'}, grouped = false] => [orderkey:bigint, linenumber:integer]
             Estimates: {rows: 60175 (822.71kB), cpu: 842450.00, memory: 0.00, network: 0.00}
             linenumber := tpch:linenumber (1:93)
             orderkey := tpch:orderkey (1:93)
```